ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/netty/NettyP2PNetwork.java

```diff
@@ -497,6 +546,42 @@ public void start() {
     };
   }
 
+  private synchronized void handleBlockAddedEvent(
+      final BlockAddedEvent event, final Blockchain blockchain) {
+    connections
+        .getPeerConnections()
+        .forEach(
+            peerConnection -> {
+              if (!isPeerConnectionAllowed(peerConnection)) {
+                peerConnection.disconnect(DisconnectReason.REQUESTED);
+              }
+            });
+  }
+
+  private boolean isPeerConnectionAllowed(final PeerConnection peerConnection) {
+    LOG.trace(
+        "Checking if connection with peer {} is permitted", peerConnection.getPeer().getNodeId());
+
+    final String localNodeId = ourPeerInfo.getNodeId().toString().substring(2);
+    final InetSocketAddress localAddress = (InetSocketAddress) peerConnection.getLocalAddress();
+    final String localHostAddress = localAddress.getHostString();
+    final int localPort = localAddress.getPort();
+    final EnodeURL localPeerEnodeURL = new EnodeURL(localNodeId, localHostAddress, localPort);
+
+    final String remoteNodeId = peerConnection.getPeer().getNodeId().toString().substring(2);
+    final InetSocketAddress remoteAddress = (InetSocketAddress) peerConnection.getRemoteAddress();
+    final String remoteHostAddress = remoteAddress.getHostString();
+    final int remotePort = remoteAddress.getPort();
+    final EnodeURL remotePeerEnodeURL = new EnodeURL(remoteNodeId, remoteHostAddress, remotePort);
+
+    return nodePermissioningController
+        .map(
+            c ->
+                c.isPermitted(localPeerEnodeURL, remotePeerEnodeURL)
```

Comment:
Something feels wrong about this check. Don't think it should be `||`. Wondering if the directional part of the rules makes sense or not, or whether we need to track the direction a connection was established in?